TCMSApp/src/client/app/defects/defects.controller.js

```diff
@@ -5,15 +5,29 @@
         .module('app.defects')
         .controller('DefectsController', DefectsController);
 
-    DefectsController.$inject = ['$scope', 'logger', '$uibModal'];
+    DefectsController.$inject = ['$scope', 'logger', '$uibModal', 'getDefects'];
     /* @ngInject */
-    function DefectsController($scope, logger, $uibModal) {
+    function DefectsController($scope, logger, $uibModal, getDefects) {
         var vm = this;
 
+        $scope.arrayDefects = getDefects.fakeDefect(10);
+        var store = $scope.arrayDefects[0];
```

Comment:
why don't you just make `var store` the `$scope.store` then you would not need the check function, ng-show and ng-repeat at the right block. (you can just display current defect there from $scope.store)